integration-tests/tests/src/tests/schema.ts

```diff
@@ -17,34 +17,47 @@
 ////////////////////////////////////////////////////////////////////////////
 
 import { expect } from "chai";
-import { Realm } from "realm";
+import { openRealmBefore } from "../hooks";
+
+interface Test {
+  primary?: number;
+  value?: number;
+}
 
 describe("Realm schema", () => {
   describe("Default property values", () => {
-    it("can take a function as a default property value", () => {
-      interface Test {
-        dynamic?: number;
-      }
-
-      const realm = new Realm({
-        schema: [
-          {
-            name: "Test",
-            properties: {
-              dynamic: {
-                type: "int",
-                default: () => 42,
-              },
+    openRealmBefore({
+      schema: [
+        {
+          name: "Test",
+          primaryKey: "primary",
+          properties: {
+            primary: {
+              type: "int",
+              default: () => 42,
+            },
+            value: {
+              type: "int",
+              default: () => 13,
             },
           },
-        ],
-      });
-
+        },
+      ],
+    });
+    it("can take a function as a default property value", function (this: RealmContext) {
+      const { realm } = this;
       const test = realm.write(() => {
         return realm.create<Test>("Test", {});
       });
 
-      expect(test.dynamic).to.equal(42);
+      expect(test.primary).to.equal(42);
+      expect(test.value).to.equal(13);
+
+      const staticTest = realm.write(() => {
+        return realm.create<Test>("Test", { primary: 0, value: 0 });
+      });
+      expect(staticTest.primary).to.equal(0);
+      expect(staticTest.value).to.equal(0);
```

Comment:
Consider introducing extra scoping blocks for each sub-test so that you can just name `staticTest` `test` (or better yet `obj` or `inserted`).

Reply:
I agree, or we could have separate `it` tests instead.

Reply:
I've split up the tests and also renamed `test` to `obj`